HW06.Rmd

```diff
@@ -1,92 +1,146 @@
+
+```{r map gene.ids to gene.symbols using ensembldb}
+#install.packages("BiocManager")
+#BiocManager::install("ensembldb")
+#BiocManager::install("EnsDb.Hsapiens.v75")
+```
+
+```{r mapping continued, cache=TRUE}
+library(ensembldb)
+library(EnsDb.Hsapiens.v75)
+ens.GBM.transcripts <- GBM.transcripts %>% 
+  mutate(gene_id = gsub(pattern = "\\..*", "", .$gene_id))
+
+map <- ensembldb::select(EnsDb.Hsapiens.v75, keys = ens.GBM.transcripts$gene_id,
+                         keytype = "TXID", columns = c("SYMBOL", "TXID"))
+
+ens.mapped_GBM <- left_join(ens.GBM.transcripts, map, by = c("gene_id" = "TXID")) %>% 
+  dplyr::select(-1) %>% 
+  dplyr::select(gene_symbol = SYMBOL, everything())
+
+ens.mapped_GBM <- ens.mapped_GBM[!duplicated(ens.mapped_GBM$gene_symbol),] #remove duplicated gene symbols
+  #these are removed instead of averaged because they simply do not correlate particularly well. 
+ens.mapped_GBM <- ens.mapped_GBM[!is.na(ens.mapped_GBM$gene_symbol),] #remove NA values
+
+```
+
+Do the t-test and make a table of the t-test results!
+
+```{r t.test for difference between a bunch of genes, cache=TRUE}
+#run this code to unload the libraries from before, it might be helpful because the select() function from dplyr might be hidden otherwise
+detach(package:EnsDb.Hsapiens.v75, unload = T)
+detach(package:ensembldb, unload = T)
+
+#add in your own gene of interest!!! 
+genes_of_interest <- c("IRX1", "OSR1", "DCHS2", "BRINP3", "TOB2P1", "FOXD1", "ZFPM2", "GLB1", "ALG5", "TRIM4", "ADARB2", "PCDHGA11", "IDH1", "EGFR", "MGMT", "TERT", "PTEN", "TP53", "RB1", "PTEN", "ATRX", "PDGFRA", "PIK3CA", "MICA", "CDKN2A", "EZH2", "BRD2")
+
+GBM.genes.of.interest <- filter(ens.mapped_GBM, gene_symbol %in% genes_of_interest)
+
+#Now perform a t-test between the H3K mutated and the wt samples. There are many ways to do this actually, you can use a for loop or you could do the tidy alternative with broom(), but the for loop is probably the easiest
+
+#reshape
+GBM.genes.of.interest<- GBM.genes.of.interest %>%
+  gather(key="sample", value="tpm", 2:46) 
+
+H3K27M<- GBM.genes.of.interest %>%
+  filter(str_detect(sample, "H3K27M")) %>%
+  mutate(sample= "H3K27M")
+
+WT<- GBM.genes.of.interest %>%
+  filter(str_detect(sample, "WT")) %>%
+  mutate(sample= "WT")
+
+twosample_GBM.genes.of.interest<- rbind(H3K27M, WT)
+
+#GBM.genes.of.interest$sample[1:23]<- "H3K27M"
+#GBM.genes.of.interest$sample[24:45]<- "WT"
+
+for (gene in genes_of_interest) {
+gene_select <- data.frame(twosample_GBM.genes.of.interest) %>% 
+  filter(gene_symbol== gene)
+print(t.test(tpm~ sample, gene_select))}
+
```

Comment:
I also mentioned this in Mingyu's homework. I cannot see which t-test correspond to which sample in the results, and I think that could be some useful information.

Reply:
I agree. I also found the results of t-test very hard to read in my own markdown file.